### PR TITLE
Add operations on daemonsets to cluster role

### PR DESCRIPTION
**Pull request by almaslennikov**
Operator fails to reconcile the MaintenanceRequest, stating it can't access daemonsets

### CODE REVIEW

**Comment by almaslennikov** [comment]
Error message:
```
2024-09-16T13:58:29.716280744Z	ERROR	controller/controller.go:316	Reconciler error	{"controller": "nodemaintenance", "controllerGroup": "maintenance.nvidia.com", "controllerKind": "NodeMaintenance", "NodeMaintenance": {"name":"my-maintenance-operation","namespace":"default"}, "namespace": "default", "name": "my-maintenance-operation", "reconcileID": "2ba1e85d-9707-4d36-9d5f-1894dd35703a", "error": "failed to update drain status. failed to get pods for deletion. [cannot delete daemonsets.apps \"nic-configuration-operator-nic-configuration-daemon\" is forbidden: User \"system:serviceaccount:maintenance-operator:maintenance-operator-controller-manager\" cannot get resource \"daemonsets\" in API group \"apps\" in the namespace \"nic-configuration-operator-system\": nic-configuration-operator-system/nic-configuration-operator-nic-configuration-daemon-79hst, cannot delete daemonsets.apps \"calico-node\" is forbidden: User \"system:serviceaccount:maintenance-operator:maintenance-operator-controller-manager\" cannot get resource \"daemonsets\" in API group \"apps\" in the namespace \"calico-system\": calico-system/calico-node-zdgct, cannot delete daemonsets.apps \"csi-node-driver\" is forbidden: User \"system:serviceaccount:maintenance-operator:maintenance-operator-controller-manager\" cannot get resource \"daemonsets\" in API group \"apps\" in the namespace \"calico-system\": calico-system/csi-node-driver-7m8b5, cannot delete daemonsets.apps \"kube-proxy\" is forbidden: User \"system:serviceaccount:maintenance-operator:maintenance-operator-controller-manager\" cannot get resource \"daemonsets\" in API group \"apps\" in the namespace \"kube-system\": kube-system/kube-proxy-8v5c2]"}
```

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -44,6 +44,17 @@ rules:
   - list
   - patch
   - update
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  verbs:
+  - delete
```
Comment (adrianchiris) [inline config/rbac/role.yaml:52]: can you only set get list and watch ?

we dont need patch update and delete

Comment (almaslennikov) [reply config/rbac/role.yaml]: Done

**Review by adrianchiris — APPROVED** [review]
LGTM

**Comment by adrianchiris** [comment]
Thx @almaslennikov will merge once CI is green

**Pull request merged**